File: bin/bootstrap/install-coffeeshop.ts
import { execSync } from "node:child_process";
import { log, which } from "./utils.js";

export function installCoffeeshop(): boolean {
  if (which("coffeeshop")) {
    log("  ✓", "Coffee Shop installed");
    return true;
  }

  log("◆", "Installing Coffee Shop...");
  try {
    execSync("npm install -g coffeeshop", { stdio: "inherit" });
    log("  ✓", "Coffee Shop installed globally");
    return true;
  } catch {
    log(
      "  ✗",
      "Failed to install Coffee Shop. Install it manually: npm install -g coffeeshop",
    );
    return false;
  }
}

File: cli/crates/talentclaw/src/bootstrap/install_coffeeshop.rs
use crate::helpers;
use std::process::Command;

/// Check if Coffee Shop CLI is available. Install if missing.
pub fn check_or_install() -> bool {
    if which::which("coffeeshop").is_ok() {
        return true;
    }

    helpers::log("◆", "Coffee Shop CLI not found. Installing...");

    let status = Command::new("npm")
        .args(["install", "-g", "coffeeshop"])
        .status();

    match status {
        Ok(s) if s.success() => {
            helpers::log("✓", "Coffee Shop CLI installed");
            true
        }
        _ => {
            helpers::log("✗", "Failed to install Coffee Shop CLI. Install manually: npm install -g coffeeshop");
            false
        }
    }
}
